bedboss/bedstat/bedstat.py

```diff
@@ -0,0 +1,229 @@
+    :param str bedfile: a full path to bed file to process
+    :param str bigbed: a full path to bigbed
+    :param str bedbase_config: a path to the bedbase configuration file
+    :param str open_signal_matrix: a full path to the openSignalMatrix
+        required for the tissue specificity plots
+    :param str outfolder: folder for storing results of the pipeline
+    :param str genome: genome assembly of the sample
+    :param str ensdb: a full path to the ensdb gtf file required for genomes
+        not in GDdata
+    :param str sample_yaml: a yaml config file with sample attributes to pass
+        on more metadata
+        into the database
+    :param bool just_db_commit: whether just to commit the JSON to the database
+    :param bool no_db_commit: whether the JSON commit to the database should be
+        skipped
+    :param bool force_overwrite: whether to overwrite the existing record
+    :param pm: pypiper object
+    """
+    outfolder_stats = os.path.join(outfolder, "output", "bedstat_output")
+    bbc = bbconf.BedBaseConf(config_path=bedbase_config, database_only=True)
+    # outfolder_stats = bbc.get_bedstat_output_path()
+
+    bed_digest = md5(open(bedfile, "rb").read()).hexdigest()
+    bedfile_name = os.path.split(bedfile)[1]
+
+    fileid = os.path.splitext(os.path.splitext(bedfile_name)[0])[0]
+    outfolder = os.path.abspath(os.path.join(outfolder_stats, bed_digest))
+    json_file_path = os.path.abspath(os.path.join(outfolder, fileid + ".json"))
+    json_plots_file_path = os.path.abspath(
+        os.path.join(outfolder, fileid + "_plots.json")
+    )
+    bed_relpath = os.path.relpath(
+        bedfile,
+        os.path.abspath(os.path.join(outfolder_stats, os.pardir, os.pardir)),
+    )
+    bigbed_relpath = os.path.relpath(
+        os.path.join(bigbed, fileid + ".bigBed"),
+        os.path.abspath(os.path.join(outfolder_stats, os.pardir, os.pardir)),
+    )
+    if not just_db_commit:
+        if force_overwrite:
+            new_start = True
+
+        if not pm:
+            pm = pypiper.PipelineManager(
+                name="bedstat-pipeline",
+                outfolder=outfolder,
+            )
+
+        # run Rscript
+        rscript_path = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            "bedstat",
+            "tools",
+            "regionstat.R",
+        )
+        assert os.path.exists(rscript_path), FileNotFoundError(
+            f"'{rscript_path}' script not found"
+        )
+        command = (
+            f"Rscript {rscript_path} --bedfilePath={bedfile} "
+            f"--fileId={fileid} --openSignalMatrix={open_signal_matrix} "
+            f"--outputFolder={outfolder} --genome={genome} "
+            f"--ensdb={ensdb} --digest={bed_digest}"
+        )
+
+        pm.run(cmd=command, target=json_file_path)
+
+    # now get the resulting json file and load it into Elasticsearch
```

Comment:
Are we using elasticsearch?

Reply:
Unfortunately, I can't answer this question. Is it artifact, that we forgot to remove?  @xuebingjie1990 

https://github.com/databio/bedstat/blame/07f16610dfc4a3cd8330ce38a1f827e4a2fd5f32/pipeline/bedstat.py#LL165C18-L168C29